### PR TITLE
docs(styleguide) update numbering rule and all rule formatting

### PR DESCRIPTION
**Pull request by h3xar0n**
**NOTE**: Please read the CONTRIBUTING.md guidelines before submitting your patch,
and ensure you followed them all:

https://github.com/Kong/docs.konghq.com/blob/master/CONTRIBUTING.md

### Summary

Changes the rule for numbering to reflect what is more consistent and easy to read. After examining many different contexts in our docs and exchanging feedback with a copyeditor, using digits instead of written-out numbers seemed like the better course.

There were also some lingering inconsistencies, both in rules and in examples, that are updated.

### Full changelog

* Alters **numbering** rules to use digits instead of written-out words.
* Fixes punctuation, formatting, and sentence structure for many rules.
* Enforces consistency with the latest rules, especially for entity names in bold text.

### Issues resolved

Fix #XXX

### Checklist:

- [x] [Commit message & atomicity](https://github.com/Kong/docs.konghq.com/blob/master/CONTRIBUTING.md#commit-atomicity) checked
- [x] Documentation 
- [x] [Adheres to Kong style guide](https://github.com/Kong/docs.konghq.com/blob/master/STYLEGUIDE.md)
- [x] Spellchecked my updates
- [x] Tagged "Team Docs" as reviewers
- [x] Ready to be merged

### CODE REVIEW

**Comment by levfishbluefish** [comment]
Hmm I have mixed feelings about the numbering, but because of the nature of our docs I agree that it could get too pedantic to dictate which contexts get digits vs words so I'll agree to just using digits.

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
+All upper case; if referring to an entity, use **Services** and **Routes** 
+instead.
 
 ### Basic Auth, Basic Authentication
-Use proper case, abbreviate if the context permits. Only use lowercase with a 
-hyphen in code/config samples
+Use proper case; abbreviate if the context permits. Only use lowercase (with a 
+hyphen) in code or configuration samples.
 
 * Bad: Options include LDAP Auth and basic auth.
```
Comment (levfishbluefish) [inline STYLEGUIDE.md]: Make this italics

**Comment by levfishbluefish** [comment]
wondering if our use of italics and bolds in the examples follows our own style guide rules🤔

File: STYLEGUIDE.md [hunk 2]
```diff
@@ -188,106 +200,112 @@ For multiple lines of code or an example command, use a
...
-[bold text](#bold-text)
+[bold text](#bold-text).
 
 ### Kong
-Use proper case
+Use proper case.
 
 ### Kong Admin
-Use Admin instead
+Use [**Admin** or **admin**](#admin-admin) instead.
```
Comment (levfishbluefish) [inline STYLEGUIDE.md]: This link was confusing, would it be better to say "Use **Admin** or **admin** (See #admin-admin)

Comment (h3xar0n) [reply STYLEGUIDE.md]: How's this?

Use **Admin** or **admin** (see [Admin, admin](#admin-admin)) instead.

```suggestion
Use **Admin** or **admin** (see [Admin, admin](#admin-admin)) instead.
```

File: STYLEGUIDE.md (hunk 1 above)
Comment (h3xar0n) [reply STYLEGUIDE.md]: (Not sure what "this" refers to--just _configuration_?)

**Comment by h3xar0n** [comment]
> wondering if our use of italics and bolds in the examples follows our own style guide rules🤔

The bold text is used correctly to refer to entities, specific names, etc. Any specific case in mind that isn't accounted for in the bulleted list?

I think our ad hoc use of italics is for _emphasis_, which is pretty normal. Do you think it would make sense to create a rule for italics?

**Comment by h3xar0n** [comment]
> Hmm I have mixed feelings about the numbering, but because of the nature of our docs I agree that it could get too pedantic to dictate which contexts get digits vs words so I'll agree to just using digits.

I've seen other guides come to this concession. If we go the other route and insist on writing things out, we will definitely be correcting every PR. For technical material or situations with lots of numbering, digits will probably cause the least friction.

File: STYLEGUIDE.md (hunk 1 above)
Comment (levfishbluefish) [reply STYLEGUIDE.md]: oh sorry, bad github linking - the "Bad" in this example is the only "Bad" not in italics

**Pull request merged**